src/Oracles/Config/Setting.hs

```diff
@@ -51,6 +52,13 @@ data Setting = BuildArch
              | GmpLibDir
              | IconvIncludeDir
              | IconvLibDir
+             | Prefix
+             | BinDir
+             | LibDir
+             | Install
+             | InstallData
+             | InstallProgram
+             | InstallDir
```

Comment:
@izgzhen Yes, over the course of the project I've also come to the conclusion that this is an unnecessary burden. 

OK, disregard this issue then. I'll go through the code and will see if I can reorder the lists in a more easy-to-maintain/understand order. If I find time to do this before this PR, you might have to do some rebasing, but hopefully nothing major.